sqflite/lib/src/adapter.dart

```diff
@@ -6,133 +6,33 @@ library jaguar_query_sqflite.src;
 import 'dart:async';
 
 import 'package:jaguar_query/jaguar_query.dart';
-import 'package:jaguar_query_sqflite/composer.dart';
+import 'package:jaguar_query_sqflite/src/connection.dart';
 import 'package:sqflite/sqflite.dart' as sqf;
 
-class SqfliteAdapter implements Adapter<sqf.Database> {
-  sqf.Database _connection;
```

Comment:
Currently in v3 I'm extending `SqfliteAdapter ` to be able to switch database on the fly for the all application. 
To do so I'm overriding the `connection` getter to return the current selected database. 

With your current changes I not be able to do that anymore because:
- you have now `SqfConn ` managing the database
- you remove constructor with database as parameter to instead have an `open` method that can't override his database

We have two choice here: 
1) Make changes to allow people to override stuff to be able to switch database on the fly if they need
2) Integrate a switchDatabase method into `SqfliteAdapter`

From previous discussion with the maintainer the solution 2 was not acceptable as Adapter should be immutable (you can see more about the discussion here https://github.com/Jaguar-dart/jaguar_orm/pull/72). 

So I guess we have to go for 1 like I did previously

Reply:
Here you can find my custom adapter I use with the current v3:
```

class KiwiSqfliteAdapter extends SqfliteAdapter {
  sqf.Database _connection;
  final _log = Logger('SQL');

  KiwiSqfliteAdapter.fromConnection(sqf.Database connection) : super.fromConnection(connection) {
    _connection = connection;
  }

  /// Change connection to the database.
  Future<void> switchDatabase(sqf.Database connection) async {
    if(_connection.path == connection.path) {
      //same database, do nothing
      return;
    }

    if (_connection != null && _connection.isOpen) {
      await close();
    }
    _connection = connection;
  }

  @override
  Future<Map> findOne(Find st) async {
    if(kSqliteDebug) {
      _log.finest(composeFind(st));
    }
    final result = await super.findOne(st);
    if(kSqliteDebug) {
      _log.finest(result);
    }
    return result;
  }

  @override
  Future<List<Map>> find(Find st) async {
    if(kSqliteDebug) {
      _log.finest(composeFind(st));
    }
    final results = await super.find(st);
    if(kSqliteDebug) {
      _log.finest(results.toString());
    }
    return results;
  }

  @override
  Future<int> remove(Remove st) {
    if(kSqliteDebug) {
      _log.finest(composeRemove(st));
    }
    return super.remove(st);
  }

  @override
  Future<int> update(Update st) {
    if(kSqliteDebug) {
      _log.finest(composeUpdate(st));
    }
    return super.update(st);
  }

  @override
  Future<T> insert<T>(Insert st) {
    if(kSqliteDebug) {
      _log.finest(composeInsert(st));
    }
    return super.insert(st);
  }

  @override
  Future<void> insertMany<T>(InsertMany st) {
    if(kSqliteDebug) {
      _log.finest(composeInsertMany(st));
    }
    return super.insertMany(st);
  }

  @override
  Future<void> updateMany(UpdateMany st) {
    if(kSqliteDebug) {
      _log.finest(composeUpdateMany(st));
    }
    return super.updateMany(st);
  }

  @override
  Future<T> upsert<T>(Upsert st) {
    if(kSqliteDebug) {
      _log.finest(composeUpsert(st));
    }
    return super.upsert(st);
  }

  @override
  Future<void> upsertMany<T>(UpsertMany st) {
    if(kSqliteDebug) {
      _log.finest(composeUpsertMany(st));
    }
    return super.upsertMany(st);
  }

  @override
  Future<void> dropDb(DropDb st) {
    if(kSqliteDebug) {
      _log.finest(composeDropDb(st));
    }
    return super.dropDb(st);
  }

  @override
  Future<void> dropTable(Drop st) {
    if(kSqliteDebug) {
      _log.finest(composeDrop(st));
    }
    return super.dropTable(st);
  }

  @override
  Future<void> createDatabase(CreateDb st) {
    if(kSqliteDebug) {
      _log.finest(composeCreateDb(st));
    }
    return super.createDatabase(st);
  }

  @override
  Future<void> createTable(Create st) {
    if(kSqliteDebug) {
      _log.finest(composeCreate(st));
    }
    return super.createTable(st);
  }

  @override
  sqf.Database get connection => _connection;

  @override
  Future<void> alter(Alter st) {
    if(kSqliteDebug) {
      _log.finest(composeAlter(st));
    }
    return super.alter(st);
  }
}

```
The key concept here is:
```
  @override
  sqf.Database get connection => _connection;
```
As it's used by the super class it will take the current database instance allowing me to switch to any database on the fly

Reply:
As I see, you still have this method: sqf.Database get connection => _connection.connection;

You can still use it to change the connection?

Reply:
How ? Because as you move the `sqf.Database ` inside `SqfConn` I have no way to overriding it. It always be the instance inside `SqfConn` that will be used for making request and I have no way to change that with the current implementation.